### ISSUE TITLE
v8.3.0 is not proxying prometheus requests correctly

### ISSUE THREAD

**Issue by James-REANNZ** [grafana/grafana#42767]
**What happened**:

After updating from v8.2.5 to 8.3.0, all the panels on our dashboards stopped working. However, this only affected dashboards that were loaded after the the update.

**What you expected to happen**:

The update should not have broken anything.

**How to reproduce it (as minimally and precisely as possible)**:

1. Have a prometheus datasource that relies on HTTP Basic Auth proxying
2. Upgrade Grafana to 8.3.0

**Anything else we need to know?**:

We use HTTP Basic Auth in front of our prometheus instances with the same username/password as the HTTP Basic auth used for Grafana itself.

Some investigation shows that the Grafana endpoint being queried has changed from `/api/datasources/proxy/:id` to `/api/ds/query`.

**Environment**:
- Grafana version: 8.3.0
- Data source type & version: Prometheus 2.6/2.27
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: OSX 11 / Chrome

**Comment by dns2utf8** [comment]
There is a new Datatype: `QueryMetricsV2` the datasourceId is now inside that request object and no longer in the URL
Are there any docs describing that API?

**Comment by aminvakil** [comment]
Just faced this upgrading to 8.3.1

**Comment by thutmacher** [comment]
8.3.1 didn't solve the problem for me. I downgraded grafana to 8.2.5.

**Comment by kukumber** [comment]
> 8.3.1 didn't solve the problem for me. I downgraded grafana to 8.2.5.

You have to upgrade to 8.2.7 because of [CVE-2021-43798](https://nvd.nist.gov/vuln/detail/CVE-2021-43798)

**Comment by thutmacher** [comment]
> > 8.3.1 didn't solve the problem for me. I downgraded grafana to 8.2.5.
> 
> You have to upgrade to 8.2.7 because of [CVE-2021-43798](https://nvd.nist.gov/vuln/detail/CVE-2021-43798)

True. I'm on a mac using brew and there is no 8.2.7 in the history https://github.com/Homebrew/homebrew-core/commits/c5396a2d19b66c585a86fc5e2f11bf8891a3ad40/Formula/grafana.rb

**Comment by alexbilevskiy** [comment]
As a workaround, switching prometheus access mode from "server" to "browser" in datasource settings fixed the issue for me

**Comment by James-REANNZ** [comment]
Just tested the 8.3.3 release which I assume contains #42728. The issue is still present.

**Comment by ivanahuckova** [comment]
@James-REANNZ just to confirm, your Prometheus data source instance in Grafana is used with **Basic Auth** toggled and filled out **Basic Auth Details**? Or are you using **oauthPassThru**? I suspect it is the  **oauthPassThru** and this is the same issue as https://github.com/grafana/grafana/issues/43415 because I couldn't reproduce the issue with  **Basic Auth**, but I just want to make sure.

**Comment by kubicgruenfeld** [comment]
We are using Prometheus datasources with basic auth configured and are getting 401 errors since 8.3.x versions 8.2.x works fine though with same configuration.

it throws errors for query_range requests.

```
user \"username_removed\": password mismatch, client: 5.6.7.8, server: cortex.example.com, request: \"POST /prometheus/api/v1/query_range HTTP/1.1\", host: \"cortex.example.com
```

other requests are working fine though and basic auth is done from the same nginx in front of a cortex deployment:

```
"5.6.7.8 - username_removed [04/Jan/2022:09:58:42 +0000] \"GET /prometheus/api/v1/label/__name__/values?start=1641286723&end=1641290323 HTTP/1.1\" 200 81 \"-\" \"Grafana/8.3.3\" 831 0.011 [cortex-query-frontend-80] [] 1.2.3.4:80 81 0.008 200 e1de972fe65606ab3524efbb89e3fdf8\n"
```

**Comment by kubicgruenfeld** [comment]
ok, so in our case the problem was that the datasources were provisionied with basicAuthPassword which is deprecated since a while. We now moved to secureJsonData and it works again. There seems so indication in the release notes that this is intended and so it seems still a bug. I just saw the log line `[Deprecated] the use of basicAuthPassword field is deprecated. Please use secureJsonData.basicAuthPassword` and gave it a shot.

**Comment by James-REANNZ** [comment]
@ivanahuckova 

> @James-REANNZ just to confirm, your Prometheus data source instance in Grafana is used with **Basic Auth** toggled and filled out **Basic Auth Details**? Or are you using **oauthPassThru**? I suspect it is the **oauthPassThru** and this is the same issue as #43415 because I couldn't reproduce the issue with **Basic Auth**, but I just want to make sure.

It's neither, I think. Basic Auth is enabled, but previously Grafana would use the Basic (HTTP) Auth details provided to it to authenticate with the datasource.

**Comment by ivanahuckova** [comment]
@James-REANNZ Are you provisioning data source? If so, have you also checked if your `basicAuthPassword` is under `secureJsonData`?

**Comment by James-REANNZ** [comment]
Looking closer at our configuration, neither Basic Auth or oauthPassThru are enabled for the datasources. We are using the auth proxy configuration for Grafana itself and in 8.2.7 Grafana forwards the same Basic Auth credentials to the datasource (which has been configured with the same set of credentials). Grafana 8.2.7 appears to copy the headers from the original request to the request to the datasource (which makes sense for a proxy), 8.3.x does not.

**Comment by hasso** [comment]
Same happens here – Basic Auth is configured in UI, no oauth passthru, no provisioning.

**Comment by ivanahuckova** [comment]
@hasso I wasn't able to reproduce this - when testing   Basic Auth configured in UI, the Auth header is correctly sent. Could you try to create new Prometheus data source with same basic auth + url + any other settings you have to see if it works. 
Also, in next 8.3.4 there are coming 2 new fixes that could possible fix your issues:
- respecting http method https://github.com/grafana/grafana/pull/43341
- forward oauth token/headers https://github.com/grafana/grafana/pull/43686

**Comment by hasso** [comment]
OK. Newly created entry works, the one created with older version of Grafana doesn't. My guess it's separate issue and related to changes how datasource data is stored in database? These datasources are identical in UI, but the first one is created in 8.3.3, another one in much older version, 6.x probably.

```
sqlite> select * from data_source;
1|1|2|prometheus|Prometheus|proxy|https://my.prometheus.host/||||1|prometheus||1|{"httpMethod":"GET"}|2022-01-08 06:00:00|2022-01-08 07:00:00|0|{"basicAuthPassword":"<string>"}|0|<string>
sqlite> select * from data_source;
1|1|9|prometheus|Prometheus|proxy|https://my.prometheus.host/||||1|prometheus|<string>|1|{"httpMethod":"GET","keepCookies":[]}|2019-01-16 00:00:00|2022-01-07 00:00:00|0|{}|0|<string>
```

**Comment by ivanahuckova** [comment]
@grafana/plugins-platform-backend do you have any thoughts why prometheus instance created in v6 wouldn't work now? We have changed where we store `basicAuthPassword` to `secureJsonData.basicAuthPassword`, but wouldn't this be updated automatically at some point if the data source was created in the UI?

**Comment by wbrowne** [comment]
This is related to https://github.com/grafana/grafana/issues/37626. See comment https://github.com/grafana/grafana/issues/37626#issuecomment-900095760

**Comment by ivanahuckova** [comment]
Ok, so if you've set up your data source be before v6.2 you need to go to data source config, re-enter the password or basic auth password and save the data source. More in https://grafana.com/docs/grafana/latest/installation/upgrading/#ensure-encryption-of-data-source-secrets. Also fix for oatuthPassTrough will be part of 8.3.4 release. 

Closing the issue now. If you encounter problem even after 8.3.4 update, please open the new issue.
